### PR TITLE
Use NETSdk projects for Desktop framework templates

### PR DESCRIPTION
**Pull request by KevinRansom**
The existing .NET Framework templates, Console, Library and Tutorial (.NET Framework) use the legacy project format.  That doesn't allow use of PackageReference or the more straightforward project format.

This PR fixes that, it also ensures we don't need to deploy an FSharp.Core and ValueTuple nuget package with the VS IDE to support those projects.

image

/cc @brettfo , @cartermp 

Where has the AssemblyInfo.fs file gone?
NetSDK projects have a Package properties page:
image

How do I change the target framework?
image

How do I add an App.config if required?
Right click on the project in solution explorer, select Add New Item,
image

### CODE REVIEW

File: vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj [hunk 1]
```diff
@@ -1,65 +1,31 @@
...
+$if$ ($targetframeworkversion$ == 4.5)    <TargetFramework>net45</TargetFramework> $endif$
+$if$ ($targetframeworkversion$ == 4.6)    <TargetFramework>net46</TargetFramework> $endif$
+$if$ ($targetframeworkversion$ == 4.6.1)    <TargetFramework>net461</TargetFramework> $endif$
+$if$ ($targetframeworkversion$ == 4.6.2)    <TargetFramework>net462</TargetFramework> $endif$
+$if$ ($targetframeworkversion$ == 4.7)    <TargetFramework>net47</TargetFramework> $endif$
+$if$ ($targetframeworkversion$ == 4.7.1)    <TargetFramework>net471</TargetFramework> $endif$
+$if$ ($targetframeworkversion$ == 4.7.2)    <TargetFramework>net472</TargetFramework> $endif$
+$if$ ($targetframeworkversion$ == 4.8)    <TargetFramework>net48</TargetFramework> $endif$
+
     <AssemblyName>$safeprojectname$</AssemblyName>
```
Comment (cartermp) [inline vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj:12]: Should this just get removed? It is for .NET SDK templates from the CLI

File: vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj [hunk 2]
```diff
@@ -1,65 +1,31 @@
...
-  <PropertyGroup Condition=" '$(FSharpTargetsPath)' == '' AND Exists('$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets') ">
-    <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
+    <AssemblyTitle>$projectname$</AssemblyTitle>
+    <AssemblyCompany>$registeredorganization$</AssemblyCompany>
+    <AssemblyProduct>$projectname$</AssemblyProduct>
+    <AssemblyCopyright>Copyright © $registeredorganization$ $year$</AssemblyCopyright>
+
+    <!-- <AssemblyVersion>1.0.*</AssemblyVersion>  -->
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <AssemblyFileVersion>1.0.0.0</AssemblyFileVersion>
```
Comment (cartermp) [inline vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj:28]: This should similarly be removed, we don't set them by default in the CLI templates

File: vsintegration/ProjectTemplates/LibraryProject/Template/Library.fsproj [hunk 3]
```diff
@@ -1,61 +1,32 @@
...
-  <PropertyGroup Condition=" '$(FSharpTargetsPath)' == '' AND Exists('$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets') ">
-    <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
+    <AssemblyTitle>$projectname$</AssemblyTitle>
+    <AssemblyCompany>$registeredorganization$</AssemblyCompany>
+    <AssemblyProduct>$projectname$</AssemblyProduct>
+    <AssemblyCopyright>Copyright © $registeredorganization$ $year$</AssemblyCopyright>
+
+    <!-- <AssemblyVersion>1.0.*</AssemblyVersion>  -->
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <AssemblyFileVersion>1.0.0.0</AssemblyFileVersion>
```
Comment (cartermp) [inline vsintegration/ProjectTemplates/LibraryProject/Template/Library.fsproj:29]: These should be removed as well

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsproj [hunk 4]
```diff
@@ -1,66 +1,31 @@
...
-  <PropertyGroup Condition=" '$(FSharpTargetsPath)' == '' AND Exists('$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets') ">
-    <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
+    <AssemblyTitle>$projectname$</AssemblyTitle>
+    <AssemblyCompany>$registeredorganization$</AssemblyCompany>
+    <AssemblyProduct>$projectname$</AssemblyProduct>
+    <AssemblyCopyright>Copyright © $registeredorganization$ $year$</AssemblyCopyright>
+
+    <!-- <AssemblyVersion>1.0.*</AssemblyVersion>  -->
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <AssemblyFileVersion>1.0.0.0</AssemblyFileVersion>
```
Comment (cartermp) [inline vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsproj:28]: This should be removed

File: vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj (hunk 1 above)
Comment (KevinRansom) [reply vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj:12]: No ... projectname can contain characters that are not valid file name chars.

File: vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj (hunk 2 above)
Comment (KevinRansom) [reply vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj:28]: No, if we take them away we will get to answer the question, how do we set the version.

File: vsintegration/ProjectTemplates/LibraryProject/Template/Library.fsproj (hunk 3 above)
Comment (KevinRansom) [reply vsintegration/ProjectTemplates/LibraryProject/Template/Library.fsproj:29]: See above

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsproj (hunk 4 above)
Comment (KevinRansom) [reply vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsproj:28]: See above.

File: vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj (hunk 2 above)
Comment (cartermp) [reply vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj:28]: Sure, so just like the CLI then. Are you proposing we add these to the CLI as well?

File: vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj (hunk 1 above)
Comment (cartermp) [reply vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj:12]: Then why is this only done for this project and not the other?

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsproj (hunk 4 above)
Comment (KevinRansom) [reply vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsproj:28]: @cartermp , they do, but they have a big page for setting it on project properties:
image.

Comment (KevinRansom) [reply vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsproj:28]: @cartermp , you are right, the project system has the same dialog for F# projects ... noice.

File: vsintegration/ProjectTemplates/LibraryProject/Template/Library.fsproj (hunk 3 above)
Comment (KevinRansom) [reply vsintegration/ProjectTemplates/LibraryProject/Template/Library.fsproj]: Updated

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsproj (hunk 4 above)
Comment (KevinRansom) [reply vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsproj]: Updated

File: vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj (hunk 1 above)
Comment (KevinRansom) [reply vsintegration/ProjectTemplates/ConsoleProject/Template/ConsoleApplication.fsproj]: Updated

**Pull request merged**